Sources/Client/Client/API/Client+Devices.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  Client+Devices.swift
+//  StreamChatCore
+//
+//  Created by Alexey Bukhtin on 09/01/2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+// MARK: - Devices
+
+public extension Client {
+    
+    /// Add a device for Push Notifications.
+    /// - Parameters:
+    ///   - deviceToken: a device token.
+    ///   - completion: an empty completion block.
+    @discardableResult
+    func addDevice(deviceToken: Data, _ completion: @escaping Client.Completion<EmptyData> = { _ in }) -> URLSessionTask {
+        if deviceToken.isEmpty {
+            completion(.failure(.emptyDeviceToken))
+            return .empty
+        }
+        
+        return addDevice(deviceId: deviceToken.deviceToken, completion)
+    }
+    
+    /// Add a device for Push Notifications.
+    /// - Parameters:
+    ///   - deviceId: a Push Notifications device identifier.
+    ///   - completion: an empty completion block.
+    @discardableResult
+    func addDevice(deviceId: String, _ completion: @escaping Client.Completion<EmptyData> = { _ in }) -> URLSessionTask {
+        guard let currentUser = user else {
+            completion(.failure(.emptyUser))
+            return .empty
+        }
+        
+        let device = Device(deviceId)
+        
+        if currentUser.devices.contains(where: { $0.id == deviceId }) {
```

Comment:
The API endpoint ensure the device is registered and that's all you should care about when you use it. We should keep it simple on the client-side as well and drop the whole `currentUser.devices` check

Reply:
so my suggestion is to only do `return request(endpoint: .addDevice(deviceId: deviceId, currentUser), completion)` here

Reply:
done